mec/cmd/mec/main.go

```diff
@@ -47,56 +47,69 @@ var (
 	registryURL    string
 	resyncPeriod   string
 	namespace      string
+
+	mainlog        = log.RegisterScope("main", "Main function", 0)
+	loggingOptions = log.DefaultOptions()
 )
 
 func main() {
-	cmd := createCommand(os.Args[1:])
-	if err := cmd.Execute(); err != nil {
-		log.Errora(err)
+	rootCmd := createCommand(os.Args[1:])
+	_ = rootCmd.Execute()
+}
+
+func configureLogging(_ *cobra.Command, _ []string) error {
+	if err := log.Configure(loggingOptions); err != nil {
+		return err
 	}
+	return nil
 }
 
 func createCommand(args []string) *cobra.Command {
-	cmd := &cobra.Command{
-		Use: "mec [flags]",
-		Run: func(cmd *cobra.Command, args []string) {
+	rootCmd := &cobra.Command{
+		Use:               "mec [flags]",
+		PersistentPreRunE: configureLogging,
+		RunE: func(command *cobra.Command, args []string) error {
 			resyncPeriod, err := time.ParseDuration(resyncPeriod)
 			if err != nil {
-				log.Warnf("Failed to parse --resyncPeriod parameter, using default 5m: %v", err)
+				mainlog.Warnf("Failed to parse --resyncPeriod parameter, using default 5m: %v", err)
 				resyncPeriod = defaultResyncPeriod
 			}
+
 			config, err := kube.BuildClientConfig("", "")
 			if err != nil {
-				log.Errorf("Failed to BuildClientConfig(): %v", err)
+				return fmt.Errorf("failed to BuildClientConfig(): %v", err)
 			}
+
 			mrc, err := memberroll.NewMemberRollController(
 				config,
 				namespace,
 				"default",
 				resyncPeriod,
 			)
 			if err != nil {
-				log.Errorf("Failed to create MemberRoll Controller: %v", err)
+				return fmt.Errorf("failed to create MemberRoll Controller: %v", err)
 			}
+
 			ec, err := extension.NewControllerFromConfigFile(
 				"",
-				[]string{"test"},
+				[]string{namespace},
 				mrc,
 				resyncPeriod,
 			)
 			if err != nil {
-				log.Errorf("Failed to create Extension Controller: %v", err)
+				return fmt.Errorf("failed to create Extension Controller: %v", err)
 			}
 
-			p, err := ossm.NewOSSMPullStrategy(config, namespace)
+			p, err := ossm.NewOSSMPullStrategy(config /*, namespace*/)
 			if err != nil {
-				log.Errorf("Failed to create OSSMPullStrategy: %v", err)
+				return fmt.Errorf("failed to create OSSMPullStrategy: %v", err)
 			}
-			w, err := server.NewWorker(config, p, baseURL, serveDirectory)
+
+			w, err := server.NewWorker(config, p, baseURL, serveDirectory, nil)
```

Comment:
why set the `errorChan` to `nil` here? doesn't that mean we'll never log any errors?

Reply:
nope. We log the error on line 254. The channel is used primarily for unit tests so that they can inspect the errors.

Reply:
I see, yeah I get it now